### PR TITLE
Feature/cogs

### PR DESCRIPTION
**Pull request by MonkfishFillet**
Refactored two of the bot functions into cogs and allows us to:

- Reduces the size of robbot.py
- Add and remove features using the Cog commands while Robbot is running
- Organize this shit just a little bit more

Also tried to condense all of the helper classes into a single Util module - May come back to break this up eventually

### CODE REVIEW

File: robbot.py
Comment (DonChalupa) [inline robbot.py]: Since `files` refers to a single element in the list of filenames returned by `os.listdir()`, not a list itself, the name `file` would be more precise.

```suggestion
    for file in os.listdir('./cogs'):
        if file.endswith('.py'):              
```

File: robbot.py
Comment (DonChalupa) [inline robbot.py]: A more readable way of trimming the file extension would be to call `os.path.splitext(file)` and use that result in the load call. 

```suggestion
            cog_name, file_ext = os.path.splitext(file)
            bot.load_extension(f'cogs.{cog_name}')
            print(f'{cog_name} extension loaded')
```

Optionally, `file_ext` could be named `_`, which by convention denotes that it's unused - personally feel that `file_ext` is a bit more clear.

File: robbot.py
Comment (DonChalupa) [inline robbot.py]: See comments on `load_cogs()`.

```suggestion
    for file in os.listdir('./cogs'):
        if file.endswith('.py'):              
            cog_name, file_ext = os.path.splitext(file)
            bot.unload_extension(f'cogs.{cog_name}')
            print(f'{cog_name} extension unloaded')
```

File: cogs/Bartender.py
Comment (DonChalupa) [inline cogs/Bartender.py]: Possible typo

```suggestion
    @commands.command(name='drink', description='Get a nice, refreshing drink from the Bartender',aliases=['drinks', 'cocktail', 'cocktails'])
```

File: cogs/Music.py
Comment (DonChalupa) [inline cogs/Music.py]: While this cog doesn't necessarily conflict with the `Music` module on my branch, it does depend on Hydra (which we no longer have in the discord) and essentially duplicates the functionality found in the `Music` module.

Rather than include this cog, we can expand on the existing `Music` module to include the functionality it provides, i.e. providing aliases for Youtube URLs. This is an issue we should hold off on until at least this merge is finished.

```suggestion

```

File: config.json
Comment (DonChalupa) [inline config.json]:
```suggestion
        "Don't count on it.",
```

File: util/Utility.py
Comment (DonChalupa) [inline util/Utility.py]: There's no need to hardcode all of this information because we should be able to get it from calls to `guild.emojis`, `guild.members`, `guild.roles`, `guild.channels`, the `Colour` class, etc. Additionally, all of this information is subject to change, duplicating the info here means any changes to the server will require changes to this file.

Whenever this information is needed, we should get it dynamically whenever possible. There are situations where we may want to get a specific channel or user by name, and in that case the duplication of info is unavoidable (e.g., a hypothetical `foo()` command that posts a message in the "#foo" channel will have to know what channel to look for, so the channel name "#foo" will have to be stored in code or config somewhere in the process), but in other situations a lookup will avoid code changes when the server changes.

For example, if we want a user's color for embeds, instead of hardcoding their color here, we can lookup the user's roles, determine their primary role, and then get the role's color. There may be times we want to get a user by their actual identity (e.g., we want a listener that responds only to James' messages), in which case hardcoding the user ID is pretty unreadable, so a simple map of colloquial names to user IDs would be acceptable and sufficient for those cases - somewhat like the `config.json` dictionary, but just first names to IDs.

File: config.json
Comment (DonChalupa) [inline config.json]: See comments on `util/Utility.py`.

Embed colors shouldn't be hardcoded here as they are subject to change (as long as we're basing them off of roles, anyway).

A dictionary mapping colloquial names to user IDs makes sense for referring to specific users in code in a readable fashion - use the colloquial name to find the user ID from this dict, then use the ID to get the rest of the user's info.

File: config.json
Comment (DonChalupa) [inline config.json]: All of this info is in the API's `Colour` class. There should be no reason to duplicate it here.

```suggestion
    ]
```

File: robbot.py
Comment (DonChalupa) [inline robbot.py:59]: I'm not sure about this yet, but I think this might be antithetical to the purpose of cogs & extensions. If I understand them right, the idea is to allow reloading of individual cogs at runtime without affecting the rest of the cogs, e.g. `bot.reload_extension('Music')` rather than unloading and loading all of them.

See [this example](https://github.com/Rapptz/RoboDanny/blob/master/cogs/admin.py#L18) I found that I think demonstrates this point better than I can explain. Also note that he has the commands in an `Admin` cog, probably a good idea for us as well.

**Review by DonChalupa — CHANGES_REQUESTED** [review]
I still have to dig in to the `models/*`, `services/*`, etc. semantics to understand how that all works together, but here's what I'm thinking with what I've gone through so far.

File: cogs/Music.py
Comment (MonkfishFillet) [reply cogs/Music.py]: Outdated and cringe cog, removed

File: util/Utility.py
Comment (MonkfishFillet) [reply util/Utility.py]: I agree with majority of this - hard coding this isn't a good look for a scalability. 

User colors was something I wanted to be customizable for each user and not dependent of role, though it's super low priority right now and can be on the back burner. Burning this file to the ground.

**Comment by DonChalupa** [comment]
Also, please pull the latest develop branch, merge it into this feature branch, resolve conflicts, then push those fixes on this feature branch to remove conflicts.

File: cogs/Pinner.py
Comment (DonChalupa) [inline cogs/Pinner.py]: `util.Utililty` no longer exists per latest commits

```suggestion
```

**Review by MonkfishFillet — COMMENTED** [review]
Should be good to go, finally on a clean slate. Let's try to get this merged before the new year.

File: cogs/Pinner.py
Comment (DonChalupa) [inline cogs/Pinner.py]: Broken by deletion of `util.Utility`. Honestly not sure the best way to handle this, seems like there's no way around hardcoding the channel name or ID somewhere.

You could search the guild's channel list for a channel called "pins" (e.g. `channel = next(filter(lambda x: x.name == "pins", ctx.guild.channels), None)`), but then if the channel name changed (for whatever reason) you'd have to change this line. You could create a constant `PIN_CHANNEL_NAME = "pins"` either in this file or in config (which is basically what you had before). That would be better since in the event of a change, you'd only have to change it in one spot in the code.

You could search for it by the channel ID instead of the name, so the channel name can change without breaking the code, but if the channel were deleted and re-added for some reason it would still break.

I think the only way we could do it without hardcoding would be to store a `PIN_CHANNEL_ID` constant in config somewhere, then add a command to set the pin channel, e.g. .set_pin_channel, which will basically execute `PIN_CHANNEL_ID = ctx.channel.id` and write it out to config. It's most likely overkill but it is robust and avoids any hardcoding which kinda gets me off.

Any of those options sound fine to me so I'll leave it up to you.

File: cogs/Pinner.py
Comment (DonChalupa) [inline cogs/Pinner.py]: Prefer not to hardcode this here, for all the same reasons listed in the comment above, but also difficult to implement without hardcoding at all (also for the same reasons above). 

At the very least, could be moved to config as `PIN_REACTION` or something, so that if it does break it only needs to be fixed in one place and that place is clear. Could also add a command that sets this reaction if you wanted to go balls out.

File: models/Drink.py
Comment (DonChalupa) [inline models/Drink.py]: Did you intend to use `__get_ingredient_string` here instead of `_get_ingredient_string` for the name mangling? Not saying there's a problem with it or it should be changed, just want to understand if there's a particular reason for the double underscores.

File: services/Embedder.py
Comment (DonChalupa) [inline services/Embedder.py]: No longer exists after deletion of `util.Utility`.
```suggestion
from discord import Colour
```

File: services/Embedder.py
Comment (DonChalupa) [inline services/Embedder.py]:
```suggestion
        embed = Embed(title= drink._name, description='A drink for you, good Bud.', color=Colour.dark_blue())
```

File: services/Embedder.py
Comment (DonChalupa) [inline services/Embedder.py]: `Colors()` no longer exists after `util.Utility` deletion

**Review by DonChalupa — CHANGES_REQUESTED** [review]
Looked through stuff a little more. Still need to understand the interplay between cogs/models/services, that kind of paradigm isn't something I'm familiar with.

**Comment by MonkfishFillet** [comment]
I lied, music isn't working due to missing packages and also some other stuff looool

File: cogs/Pinner.py
Comment (MonkfishFillet) [reply cogs/Pinner.py]: Def agree with setting a constant in there, I'm on it. I'll see if I can do a bit of digging to find the best solution to make it dynamic but realistically I think of a solution that fits all use-cases. if we search by name and the name is changed for whatever reason, broke. If we search by id and we delete it and make a new one, broke. If we delete it and change the name, double broke. Just the nature of the beast, knowwhatimsayinmatelol

For now, I think any dynamic solution is beyond the scope of this PR so I'll set a constant and come back to it later.

File: cogs/Pinner.py
Comment (MonkfishFillet) [reply cogs/Pinner.py]: Same as above, adding constant to Pinner Class

File: models/Drink.py
Comment (MonkfishFillet) [reply models/Drink.py]: Really not much of reason - thought process was I saw no reason to keep it as a protected class if the model wasn't going to have children classes inherited from it. Making it private just seemed like it was a better fit

File: services/Embedder.py
Comment (MonkfishFillet) [reply services/Embedder.py]: Had to fiddle with this a bit - apparently there was a difference between `from discord import Colour` and `from discord,colour import Colour`.  Resolved now.

Additionally, `from discord,colour import Color` works as well - this is what I ran with because because

**Comment by MonkfishFillet** [comment]
Fixed those problems. I removed robbot-env since we can replicate it using venv and the requirment.txt, and the files generated aren't necessary on the repo

**Pull request merged**
